src/pages/VenuePage/VenuePage.tsx

```diff
@@ -172,15 +172,19 @@ const VenuePage: React.FC = () => {
 
   // useVenueAccess(venue, handleAccessDenied);
 
-  if (!user) {
-    return <Login formType="initial" />;
+  if (!venue || !venueId) {
+    return <LoadingPage />;
   }
 
   if (venueRequestStatus && !venue) {
     return <>This venue does not exist</>;
   }
 
-  if (!venue || !venueId || !profile) {
+  if (!user) {
+    return <Login formType="initial" venue={venue} />;
+  }
+
+  if (!profile) {
```

Comment:
There are 2 separate checks that render `<LoadingPage/>` why is that?

One at line 175, and one here.

Reply:
I don't think it's true

![image](https://user-images.githubusercontent.com/63194656/122253733-1c1dc100-ced5-11eb-80bb-3006f9b3535f.png)

Reply:
So, I moved the `!venue || !venueId` check before the `Login page`, because I need `venue` to be loaded before we display it. However, if you add `!profile` before `!user`, it will cause an infinite loading

Reply:
Okey, sounds reasonable.

However, looks like because of the first statement, the second one will never render, yes?

```tsx
  if (!venue || !venueId) {
    return <LoadingPage />;
  }

  if (venueRequestStatus && !venue) {
    return <>This venue does not exist</>;
  }
```

Maybe you should swap them or add some additional condition to the first one.

Reply:
👍 Fixed in https://github.com/sparkletown/sparkle/pull/1546/commits/882adbf34503dedf8195a6d981d431cbd2ca71a9